.github/workflows/test.yml

```diff
@@ -23,11 +23,16 @@ jobs:
         with:
           save-if: ${{ github.ref == 'refs/heads/main' }}
 
+      - name: Validate Cargo.lock is up-to-date
+        run: |
+          cargo check --workspace
+          git diff --exit-code
```

Comment:
Does this mean that we fail our CI pipelines if a new version has been published to crates.io?

I guess it won't happen as long as the GitHub cache is fresh, but if it expires, then we'll likely see small changes in our `Cargo.lock` file, no?

Reply:
Hmm, I haven't used rust enough to know all cases when Cargo.lock is updated.
Based on my understanding, this could fail in 2 cases - PR push or after PR merge.

If during PR push, then the PR creator should re-build the project and Cargo.toml will be updated. That is expected from PR creator I think.

If after PR merge, then this means some of the deps were updated after PR push build but before merge job. I think this is OK as it will be fixed in the next PR.

Reply:
> Hmm, I haven't used rust enough to know all cases when Cargo.lock is updated.
> Based on my understanding, this could fail in 2 cases - PR push or after PR merge.

I'm afraid of spurious and unrelated changes: I fix a comment and my PR fails because someone released a new patch version of some random crate... That will be confusing for me 🙂

I guess we'll know soon enough and then we can reconsider.